lib/src/license_detection/crc32.dart

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Specification of [CRC-32][1] is same as used in IEEE 802.3 Ethernet (Normal).
+///
+/// ```
+/// Width  : 32
+/// Poly   : 0x04c11db7
+/// Init   : 0xffffffff
+/// RefIn  : true
+/// RefOut : true
+/// ```
+///
+/// [1]: https://en.wikipedia.org/wiki/Cyclic_redundancy_check
+int crc32(List<int> bytes) {
+  var crc = 0xffffffff;
+  final length = bytes.length;
+
+  for (var i = 0; i < length; i++) {
+    crc = (crc >> 8) ^ _crc32Table[(crc & 255) ^ bytes[i]];
+  }
+
+  return (~crc).toUnsigned(32);
+}
+
+/// This table was generated using [Rosetta][1] C implementation code.
+///
+/// [1]: https://rosettacode.org/wiki/CRC-32#Implementation_2
+final _crc32Table = [
```

Comment:
Documentation comments have to document what something is:
```suggestion
/// CRC-32 table generated using [Rosetta][1] C implementation code.
///
/// [1]: https://rosettacode.org/wiki/CRC-32#Implementation_2
final _crc32Table = [
```

An alternative is to not write a documentation comment and just write a normal comment:
```suggestion
// This table was generated using Rosetta C implementation code,
// See: https://rosettacode.org/wiki/CRC-32#Implementation_2
final _crc32Table = [
```

A normal comment is useful for the reader of the code.. but not intended to be useful for consumers of the code.
It doesn't have describe what something is. A normal comment is just a comment about something.

A documentation comment has to _document what something is_.

In this case, it's clear from the variable name that this is a crc-32 table, unless we want to explain how to use such a table we probably don't need to document it. So using a normal comment to say where it came from makes sense.